MessagePack allows NilValue as map value, but Java maps don't allow null as value

Suppose the driver receives a map from some stored function, where some values as MessagePack `null`s (or `box.NULL`). In this case `DefaultMapValueConverter` will fail with an error like:

```
Caused by: java.lang.NullPointerException
	at java.util.HashMap.merge(HashMap.java:1225)
	at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.tarantool.driver.mappers.DefaultMapValueConverter.fromValue(DefaultMapValueConverter.java:24)
	at io.tarantool.driver.mappers.DefaultMapValueConverter.fromValue(DefaultMapValueConverter.java:13)
	at io.tarantool.driver.mappers.DefaultMessagePackMapper.fromValue(DefaultMessagePackMapper.java:107)
	at io.tarantool.driver.mappers.DefaultMessagePackMapper.fromValue(DefaultMessagePackMapper.java:87)
```

This happens because the map collector does not allow nulls as `Map.Entry` values.

`DefaultMapValueConverter` needs to filter out the pairs with null values.

--- a/src/main/java/io/tarantool/driver/mappers/DefaultMapValueConverter.java
+++ b/src/main/java/io/tarantool/driver/mappers/DefaultMapValueConverter.java
@@ -21,6 +21,7 @@ public class DefaultMapValueConverter implements ValueConverter<MapValue, Map<?,
     @Override
     public Map<?, ?> fromValue(MapValue value) {
         return value.map().entrySet().stream()
+                .filter(e -> !e.getValue().isNilValue())
                 .collect(Collectors.toMap(e -> mapper.fromValue(e.getKey()), e -> mapper.fromValue(e.getValue())));
     }
 }

--- a/src/test/java/io/tarantool/driver/mappers/DefaultMessagePackMapperTest.java
+++ b/src/test/java/io/tarantool/driver/mappers/DefaultMessagePackMapperTest.java
@@ -114,6 +114,17 @@ class DefaultMessagePackMapperTest {
         complexMap.get(2).put(5, "It's Wednesday");
         complexMap.get(2).put(6, "My dudes");
         assertEquals(complexMap, mapper.fromValue(mapper.toValue(complexMap)));
+
+        // map with nil values
+        MapValue nilMap = ValueFactory.newMap(
+                ValueFactory.newString("a"), ValueFactory.newInteger(123),
+                ValueFactory.newString("b"), ValueFactory.newNil(),
+                ValueFactory.newString("c"), ValueFactory.newInteger(456)
+        );
+        Map<String, Integer> notNilMap = new HashMap<>();
+        notNilMap.put("a", 123);
+        notNilMap.put("c", 456);
+        assertEquals(notNilMap, mapper.fromValue(nilMap));
     }
 
     @Test

